src/neo/Network/P2P/LocalNode.cs

```diff
@@ -74,9 +74,14 @@ private void BroadcastMessage(MessageCommand command, ISerializable payload = nu
         /// Broadcast a message to all connected nodes, namely <see cref="Connections"/>.
         /// </summary>
         /// <param name="message">The message to be broadcasted.</param>
-        private void BroadcastMessage(Message message)
+        private void BroadcastMessage(Message message) => SendToRemoteNodes(message);
+
+        private void SendToRemoteNodes(object message)
         {
-            Connections.Tell(message);
+            foreach (var connection in RemoteNodes.Keys)
```

Comment:
Can you add a comment about this optimization, in order to prevent forget it in the future and change it again because looks simpler

Reply:
Sure, please check the added comment.